src/utils/deepLinkParser.ts

```diff
@@ -80,6 +81,18 @@ export const deepLinkParser = async (url, currentAccount) => {
       };
       keey = `${mode}/${referredUser || ''}`;
     }
+
+    //if url is for purchase
+    const { type, username} = parsePurchaseUrl(url) || {};
+    console.log('type, username : ', type, username);
+    if(type && type === 'boost'){
+      routeName = ROUTES.SCREENS.ACCOUNT_BOOST;
+      params = {
+        username,
```

Comment:
accountBoostScreen is not reading navigation parameters, so this username is being discarded... read it in accountBoostScreen and pass it to InAppPurchaseContainer. The prop should be optional but if preset should override the currentAccount.name already being used in InAppPurchases.

references
https://github.com/ecency/ecency-mobile/blob/d1fa36330399c05048cd546723fefc719d690f10/src/containers/inAppPurchaseContainer.ts#L84

https://github.com/ecency/ecency-mobile/blob/d1fa36330399c05048cd546723fefc719d690f10/src/screens/accountBoost/screen/accountBoostScreen.js#L27